python/pyspark/sql/connect/client/reattach.py

```diff
@@ -200,9 +200,6 @@ def _release_all(self) -> None:
         This will send an asynchronous RPC which will not block this. The client continues
         executing, and if the release fails, server is equipped to deal with abandoned executions.
         """
-        if self._result_complete:
-            return
```

Comment:
I'd rather remove line 114 https://github.com/apache/spark/pull/42929/files#diff-42a2895f9b799478210ca059e731fdbabf84042dddef97c203f90d5c5870aa3dR114 than the check here.

Reply:
Done.